x/ecocredit/migrations/v3/patch.go

```diff
@@ -0,0 +1,190 @@
+package v3
+
+import (
+	"context"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	basketapi "github.com/regen-network/regen-ledger/api/regen/ecocredit/basket/v1"
+	api "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+// patchMigrate performs following migrations
+// - add reference id to existing projects
+// - update issuance date for credit batches
+// - update curator address for baskets
+func patchMigrate(ctx context.Context, sdkCtx sdk.Context, ss api.StateStore,
+	basketStore basketapi.StateStore, oldBatchDenomToNewDenomMap map[string]string) error {
+	if sdkCtx.ChainID() == "regen-1" {
+		return patchMainnet(ctx, ss, oldBatchDenomToNewDenomMap)
+	} else if sdkCtx.ChainID() == "regen-redwood-1" {
+		return patchRedwood(ctx, ss, basketStore, oldBatchDenomToNewDenomMap)
+	}
+
+	return nil
+}
+
+func patchMainnet(ctx context.Context, ss api.StateStore, oldBatchDenomToNewDenomMap map[string]string) error {
+	// project location -> reference-id
+	// CD-MN -> "" // TODO: add reference-id
+	// KE    -> ""
+
+	locationToReferenceIdMap := make(map[string]string)
+	locationToReferenceIdMap["CD-MN"] = ""
+	locationToReferenceIdMap["KE"] = ""
```

Comment:
Committing this change and merging this pull request so that we can tag a beta and test. If we need to make any revisions, we can do so at a later point in time. Updated the issue and added followups: https://github.com/regen-network/regen-ledger/issues/980